python/cudf/cudf/core/tools/datetimes.py

```diff
@@ -331,3 +331,81 @@ def get_units(value):
         return _unit_map[value.lower()]
 
     return value
+
+
+class DateOffset(object):
```

Comment:
Should explore if we can inherit from `pd.DateOffset` here which would handle a lot of the validation for us.